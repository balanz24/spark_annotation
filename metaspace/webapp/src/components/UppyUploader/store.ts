// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from '@vue/composition-api'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach(listener => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
