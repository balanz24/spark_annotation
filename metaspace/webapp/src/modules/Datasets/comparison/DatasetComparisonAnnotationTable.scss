.dataset-comparison-annotation-table{

  .select-btn-wrapper{
    width: 146px;
    height: 42px;
  }

  .select-btn-icon{
    position: absolute;
    right: 8px;
  }

  .full-screen-icon{
    transform: scale(0.4);
    margin: -13.5px -11.5px;
  }
  .full-screen-btn{
    padding: 0;
    margin: 0;
    height: 42px;
    width: 42px;
  }

  #annot-table:focus {
    outline: 1px solid theme('colors.primary');
    outline-offset: 1px;
  }

  .msm-header{
    min-width: 80px;
    padding: 0;
    margin: 0;
  }

  /* fix cell height and align text in the center */
  #annot-table .cell {
    height: 24px;
    display: flex;
    align-items: center;
    padding: 0 0 0 10px;
  }

  #annot-table th > .cell{
    height: 30px;
    font-size: 14px;
  }

  /* don't show long group/dataset names */
  #annot-table .cell {
    white-space: nowrap;
  }

  #annot-table::after {
    background-color: transparent;
  }

  .el-table__body tr.fdr-null > td.fdr-cell, .fdr-legend.fdr-null {
    @apply bg-blue-100;
  }

  .el-table__body tr.fdr-5 > td.fdr-cell, .fdr-legend.fdr-5, .el-table__body tr.coloc-95 > td.coloc-cell {
    background-color: #c8ffc8;
  }

  .el-table__body tr.fdr-10 > td.fdr-cell, .fdr-legend.fdr-10, .el-table__body tr.coloc-90 > td.coloc-cell {
    background-color: #e0ffe0;
  }

  .el-table__body tr.fdr-20 > td.fdr-cell, .fdr-legend.fdr-20, .el-table__body tr.coloc-80 > td.coloc-cell {
    background-color: #ffe;
  }

  .el-table__body tr.fdr-50 > td.fdr-cell, .fdr-legend.fdr-50, .el-table__body tr.coloc-50 > td.coloc-cell {
    background-color: #fff5e0;
  }

  .el-table__body tr.fdr-null.current-row > td {
    background-color: theme('backgroundColor.blue.200') !important;
  }

  .el-table__body tr.fdr-5.current-row > td {
    background-color: #8f8 !important;
  }

  .el-table__body tr.fdr-10.current-row > td {
    background-color: #bfb !important;
  }

  .el-table__body tr.fdr-20.current-row > td {
    background-color: #ffa !important;
  }

  .el-table__body tr.fdr-50.current-row > td {
    background-color: #ffe5a0 !important;
  }

  .cell-wrapper {
    width: 100%;
    display: flex;
    justify-content: space-between;
  }

  .cell-span {
    @apply tracking-wide flex;
  }

  .cell-wrapper .cell-filter-button {
    @apply absolute right-0 self-center fill-current text-blue-500 opacity-0 cursor-pointer;
    width: 16px;
    height: 16px;
    padding: 2px 4px;
  }

  .cell-wrapper:hover .cell-filter-button {
    @apply opacity-100;
  }

  .fdr-legend, .fdr-legend-header {
    padding: 5px;
    margin: 0px 0px 10px 0px;
    display: inline-block;
    float: left;
    text-align: center;
  }

  .fdr-legend.fdr-5 {
    border-radius: 5px 0px 0px 5px;
  }

  .fdr-legend.fdr-50 {
    border-radius: 0px 5px 5px 0px;
  }

  #annot-table th.fdr-cell > .cell {
    padding: 0;
  }

  .fdr-cell > .cell, .coloc-cell > .cell {
    justify-content: center;
    padding: 0 10px !important;
  }

  #annot-table .el-table__empty-block {
    min-height: 300px;
  }
  #annot-table .el-table__empty-text {
    line-height: normal;
  }

  #annot-table > .el-loading-mask {
    background-color: white;
  }

  #annot-count {
    padding-left: 5px;
  }
}

.ds-comparison-annotation-table-loading-wrapper{
  @apply flex items-center justify-center;
  height: 200px;
  width: 100%;
}

.export-pop{
  @apply p-0;
}
.export-option{
  @apply cursor-pointer p-4 m-0 select-none;

  &:hover{
    @apply bg-blue-100;
  }
}