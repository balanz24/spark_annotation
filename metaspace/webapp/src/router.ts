/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/imzml/DatasetBrowserPage'),
  DatasetEnrichmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/enrichment/DatasetEnrichmentPage'),
  SpottingProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "SpottingProjectPage" */ './modules/SpottingProject/DashboardPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),

  // Separate bundle for design docs
  DesignTOC: () => import(/* webpackChunkName: "DesignBundle" */ './design/TOCPage.vue'),
  DesignStyleGuide: () => import(/* webpackChunkName: "DesignBundle" */ './design/StyleGuidePage.vue'),
  DesignIcons: () => import(/* webpackChunkName: "DesignBundle" */ './design/IconsPage.vue'),
  DesignComponents: () => import(/* webpackChunkName: "DesignBundle" */ './design/ComponentsPage.vue'),
  DesignForms: () => import(/* webpackChunkName: "DesignBundle" */ './design/FormsPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    {
      path: '/datasets/:dataset_id/comparison',
      name: 'datasets-comparison',
      component: asyncPages.DatasetComparisonPage,
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
    { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
    { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
    { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetEnrichmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/groups', component: asyncPages.GroupsListPage },
    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignTOC, meta: { footer: true, flex: true } },
    { path: '/design/styleguide', component: asyncPages.DesignStyleGuide, meta: { footer: true, flex: true } },
    { path: '/design/icons', component: asyncPages.DesignIcons, meta: { footer: true, flex: true } },
    { path: '/design/components', component: asyncPages.DesignComponents, meta: { footer: true, flex: true } },
    { path: '/design/forms', component: asyncPages.DesignForms, meta: { footer: true, flex: true } },

    { path: '/detectability', name: 'detectability', component: asyncPages.SpottingProjectPage },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
