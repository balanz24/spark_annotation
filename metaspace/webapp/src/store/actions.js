import apolloClient from '../api/graphqlClient';
import {fetchOptionListsQuery} from '../api/metadata';
import {decodeParams} from '../modules/Filters';
import {computed} from "@vue/composition-api";
import store from "../store/index";
export default {

  async initFilterLists(context) {
    if(context.state.filterListsLoading || context.state.filterLists != null)
      return;

    context.commit('setFilterListsLoading');

    const response = await apolloClient.query({
      query: fetchOptionListsQuery,
      fetchPolicy: 'cache-first',
    });

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds)} );

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)
    context.commit('updateFilter', filter)
  },

};
