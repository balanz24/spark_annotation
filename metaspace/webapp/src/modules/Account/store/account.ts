import { Module } from 'vuex'
import { DialogType } from '../dialogs'
import router from '../../../router'
import { RawLocation } from 'vue-router/types/router'

export interface AccountState {
  dialog: DialogType | null;
  dialogCloseRedirect: RawLocation | null;
  loginSuccessRedirect: RawLocation | null;
}

interface ShowDialogOptions {
  dialog: DialogType;
  dialogCloseRedirect?: RawLocation;
  loginSuccessRedirect?: RawLocation;
}

interface HideDialogOptions {
  dialog: DialogType;
  isLoginSuccess?: boolean;
}

const account: Module<AccountState, any> = {
  namespaced: true,
  state: {
    dialog: null,
    dialogCloseRedirect: null,
    loginSuccessRedirect: null,
  },
  mutations: {
    showDialog(state, payload: DialogType | ShowDialogOptions) {
      const options: ShowDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      state.dialog = options.dialog
      if (options.dialogCloseRedirect != null) {
        state.dialogCloseRedirect = options.dialogCloseRedirect
      }
      if (options.loginSuccessRedirect != null) {
        state.loginSuccessRedirect = options.loginSuccessRedirect
      }
    },
    hideDialog(state, payload: DialogType | HideDialogOptions) {
      const options: HideDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      if (state.dialog === options.dialog) {
        state.dialog = null
        if (options.isLoginSuccess && state.loginSuccessRedirect != null) {
          router.push(state.loginSuccessRedirect)
        } else if (state.dialogCloseRedirect != null) {
          router.push(state.dialogCloseRedirect)
        }

        state.dialogCloseRedirect = null
        state.loginSuccessRedirect = null
      }
    },
  },
}

export default account
