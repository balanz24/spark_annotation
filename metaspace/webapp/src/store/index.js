import Vue from 'vue';
import Vuex from 'vuex';
Vue.use(Vuex);

import getters from './getters.js';
import mutations from './mutations.js';
import actions from './actions.js';
import {accountModule} from '../modules/Account';

const store = new Vuex.Store({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // currently selected normalization matrix
    normalization: undefined,

    // roi settings
    roiInfo: {visible: false},

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,

    currentUser: {},

    // ion image global viewer settings
    channels: [],
    mode: 'SINGLE'
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule
  }
})

export default store;
