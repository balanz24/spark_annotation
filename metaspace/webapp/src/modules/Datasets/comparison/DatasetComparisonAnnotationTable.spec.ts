import { mount } from '@vue/test-utils'
import Vue from 'vue'
import { DatasetComparisonAnnotationTable } from './DatasetComparisonAnnotationTable'
import Vuex from 'vuex'
import { sync } from 'vuex-router-sync'
import store from '../../../store'
import router from '../../../router'

describe('DatasetComparisonAnnotationTable', () => {
  const mockHandleRowChange = jest.fn((idx: number) => { return idx })
  const propsData = {
    isLoading: false,
    annotations: [
      {
        id: '2021-03-31_11h02m28s_5230',
        sumFormula: 'C10H11NO',
        ion: 'C10H11NO+Na+',
        ionFormula: 'C10H11NONa',
        database: 'HMDB',
        msmScore: 0.736404,
        rhoSpatial: 0.848534,
        rhoSpectral: 0.982476,
        rhoChaos: 0.883333,
        fdrLevel: 0.05,
        mz: 184.07324581966384,
        colocalizationCoeff: null,
        offSample: null,
        offSampleProb: null,
        isomers: [],
        isobars: [],
        countPossibleCompounds: 5,
        possibleCompounds: [],
      },
      {
        id: '2021-03-31_11h02m28s_5175',
        sumFormula: 'C23H45NO4',
        adduct: '+H',
        ion: 'C23H45NO4+H+',
        ionFormula: 'C23H46NO4',
        database: 'HMDB',
        msmScore: 0.509524,
        rhoSpatial: 0.577076,
        rhoSpectral: 0.990214,
        rhoChaos: 0.891667,
        fdrLevel: 0.05,
        mz: 400.34209613966374,
        colocalizationCoeff: null,
        offSample: null,
        offSampleProb: null,
        isotopeImages: [
          {
            mz: 400.34209613966374,
            url: 'http://storage:9000/sm-ion-images-dev/iso/2021-03-31_11h02m28s/fd53ff218cfb420f988b9fc0723b802a',
            minIntensity: 8137228,
            maxIntensity: 11610989,
            totalIntensity: 76451200,
          },
        ],
        isomers: [],
        isobars: [],
        countPossibleCompounds: 1,
        possibleCompounds: [],
      },
    ],
    onRowChange: mockHandleRowChange,
  }

  const testHarness = Vue.extend({
    components: {
      DatasetComparisonAnnotationTable,
    },
    render(h) {
      return h(DatasetComparisonAnnotationTable, { props: this.$attrs })
    },
  })

  beforeAll(() => {
    Vue.use(Vuex)
    sync(store, router)
  })

  it('it should match snapshot', async() => {
    const wrapper = mount(testHarness, { store, router, propsData })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  it('it should match total count with number of annotations', async() => {
    const wrapper = mount(testHarness, { store, router, propsData })
    await Vue.nextTick()

    expect(wrapper.find('#annot-count').text())
      .toBe(`${propsData.annotations.length} matching records`)
  })
})
