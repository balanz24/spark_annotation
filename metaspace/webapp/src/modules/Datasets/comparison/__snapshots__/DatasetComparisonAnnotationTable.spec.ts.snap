// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetComparisonAnnotationTable it should match snapshot 1`] = `
<div class="dataset-comparison-annotation-table relative" annotations="[object Object],[object Object]" onrowchange="function mockConstructor(_a) {
        return fn.apply(this, arguments);
      }">
  <div class="el-table el-table--fit el-table--striped el-table--border el-table--scrollable-x el-table--enable-row-hover el-table--enable-row-transition el-table--mini" id="annot-table" current="true" elementloadingtext="Loading results …" tabindex="1">
    <div class="hidden-columns">
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
    </div>
    <div class="el-table__header-wrapper">
      <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 520px;">
        <colgroup>
          <col name="el-table_1_column_1" width="120">
          <col name="el-table_1_column_2" width="120">
          <col name="el-table_1_column_3" width="60">
          <col name="el-table_1_column_4" width="100">
          <col name="el-table_1_column_5" width="120">
        </colgroup>
        <thead class="">
          <tr class="">
            <th colspan="1" rowspan="1" class="el-table_1_column_1     is-leaf is-sortable">
              <div class="cell">Annotation<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
            </th>
            <th colspan="1" rowspan="1" class="el-table_1_column_2     is-leaf is-sortable">
              <div class="cell">
                <div class="msm-header">Best MSM<mock-el-popover trigger="hover" placement="right">
                    <div slot-key="default">Highest MSM among the datasets.</div>
                    <div slot-key="reference"><i class="el-icon-question metadata-help-icon ml-1"></i></div>
                  </mock-el-popover>
                </div><span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span>
              </div>
            </th>
            <th colspan="1" rowspan="1" class="el-table_1_column_3     is-leaf is-sortable">
              <div class="cell">m/z<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
            </th>
            <th colspan="1" rowspan="1" class="el-table_1_column_4     is-leaf is-sortable">
              <div class="cell">Datasets #<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
            </th>
            <th colspan="1" rowspan="1" class="el-table_1_column_5 ascending  fdr-cell  is-leaf is-sortable">
              <div class="cell">
                <div class="msm-header">Best FDR<mock-el-popover trigger="hover" placement="right">
                    <div slot-key="default">Lowest FDR among the datasets.</div>
                    <div slot-key="reference"><i class="el-icon-question metadata-help-icon ml-1"></i></div>
                  </mock-el-popover>
                </div><span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span>
              </div>
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <div class="el-table__body-wrapper is-scrolling-left">
      <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 520px;">
        <colgroup>
          <col name="el-table_1_column_1" width="120">
          <col name="el-table_1_column_2" width="120">
          <col name="el-table_1_column_3" width="60">
          <col name="el-table_1_column_4" width="100">
          <col name="el-table_1_column_5" width="120">
        </colgroup>
        <tbody>
          <tr class="el-table__row fdr-5 coloc-50">
            <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
              <div class="cell">
                <div class="cell-wrapper">
                  <mock-el-popover trigger="hover" placement="right" class="cell-span">
                    <div slot-key="default">
                      <div class="leading-5 py-2 px-3 font-normal content text-sm text-left">
                        <!---->
                        <!---->
                        <ul class="p-0 list-none leading-7">
                          <!---->
                        </ul>
                        <!---->
                        <!---->
                      </div>
                    </div>
                    <div slot-key="reference"><span>[C<sub class="leading-none">10</sub>H<sub class="leading-none">11</sub>NO + Na]⁺</span>
                      <!---->
                      <!---->
                    </div>
                  </mock-el-popover> <svg data-icon="filter" class="cell-filter-button"></svg>
                </div>
              </div>
            </td>
            <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
              <div class="cell">0.736</div>
            </td>
            <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
              <div class="cell">184.0732</div>
            </td>
            <td rowspan="1" colspan="1" class="el-table_1_column_4  ">
              <div class="cell"></div>
            </td>
            <td rowspan="1" colspan="1" class="el-table_1_column_5  fdr-cell">
              <div class="cell"><span>5%</span></div>
            </td>
          </tr>
          <tr class="el-table__row el-table__row--striped fdr-5 coloc-50">
            <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
              <div class="cell">
                <div class="cell-wrapper">
                  <mock-el-popover trigger="hover" placement="right" class="cell-span">
                    <div slot-key="default">
                      <div class="leading-5 py-2 px-3 font-normal content text-sm text-left">
                        <!---->
                        <!---->
                        <ul class="p-0 list-none leading-7">
                          <!---->
                        </ul>
                        <!---->
                        <!---->
                      </div>
                    </div>
                    <div slot-key="reference"><span>[C<sub class="leading-none">23</sub>H<sub class="leading-none">45</sub>NO<sub class="leading-none">4</sub> + H]⁺</span>
                      <!---->
                      <!---->
                    </div>
                  </mock-el-popover> <svg data-icon="filter" class="cell-filter-button"></svg>
                </div>
              </div>
            </td>
            <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
              <div class="cell">0.510</div>
            </td>
            <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
              <div class="cell">400.3421</div>
            </td>
            <td rowspan="1" colspan="1" class="el-table_1_column_4  ">
              <div class="cell"></div>
            </td>
            <td rowspan="1" colspan="1" class="el-table_1_column_5  fdr-cell">
              <div class="cell"><span>5%</span></div>
            </td>
          </tr>
          <!---->
        </tbody>
      </table>
      <!---->
      <!---->
    </div>
    <!---->
    <!---->
    <!---->
    <!---->
    <div class="el-table__column-resize-proxy" style="display: none;"></div>
  </div>
  <div class="flex justify-between items-start mt-2">
    <div>
      <mock-el-pagination total="2" pagesize="15" pagesizes="15,20,25,30" currentpage="1" layout="prev,pager,next,sizes"></mock-el-pagination>
      <div id="annot-count" class="mt-2"><b>2</b> matching records</div>
      <div class="mt-2">
        <div class="fdr-legend-header">FDR levels:</div>
        <div class="fdr-legend fdr-5">5%</div>
        <div class="fdr-legend fdr-10">10%</div>
        <div class="fdr-legend fdr-20">20%</div>
        <div class="fdr-legend fdr-50">50%</div>
      </div>
    </div>
    <div class="flex w-full items-center justify-end flex-wrap">
      <mock-el-popover class="mt-1">
        <div slot-key="default">
          <div class="cursor-pointer select-none">
            <div><i class="el-icon-check"></i><span>Annotation</span></div>
            <div><i class="el-icon-check invisible"></i><span>Adduct</span></div>
            <div><i class="el-icon-check"></i><span>Best MSM</span></div>
            <div><i class="el-icon-check"></i><span>m/z</span></div>
            <div><i class="el-icon-check"></i><span>Datasets #</span></div>
            <div><i class="el-icon-check invisible"></i><span>Isomers</span></div>
            <div><i class="el-icon-check invisible"></i><span>Isobars</span></div>
            <div><i class="el-icon-check invisible"></i><span>Max Intensity</span></div>
            <div><i class="el-icon-check invisible"></i><span>Molecules</span></div>
            <div><i class="el-icon-check"></i><span>Best FDR</span></div>
          </div>
        </div>
        <div slot-key="reference"><button type="button" class="el-button el-button--default select-btn-wrapper relative">
            <!---->
            <!----><span>Columns<i class="el-icon-arrow-down select-btn-icon"></i></span></button></div>
      </mock-el-popover>
      <mock-el-popover popper-class="export-pop" class="select-btn-wrapper ml-2 mt-1">
        <div slot-key="default">
          <p class="export-option">Annotations table</p>
          <p class="export-option">Pixel intensities</p>
        </div>
        <div slot-key="reference">
          <div><button type="button" class="el-button el-button--default select-btn-wrapper relative" width="146" height="42">
              <!---->
              <!----><span>Export to CSV<i class="el-icon-arrow-down select-btn-icon"></i></span></button></div>
        </div>
      </mock-el-popover>
      <div class="ml-2 mt-1"><button type="button" class="el-button el-button--default full-screen-btn">
          <!---->
          <!----><span><svg data-icon="exit_full_screen" class="full-screen-icon"></svg></span></button></div>
    </div>
  </div>
</div>
`;
